### PR TITLE
fix(runtime): interface clarity

### PR DESCRIPTION
**Pull request by EisenbergEffect**
This PR renames the following interfaces:

* `IView` => `INodeSequence`
* `IVisual` => `IView`
* `IVisualFactory` => `IViewFactory`
* `IRenderSlot` => `IViewSlot`
* `IViewOwner` => `IRenderable`

Concrete implementations have been renamed with the same naming scheme. Properties and parameters have also been renamed where appropriate. Additionally, the `bindable` and `attachable` arrays have been pluralized.

This restores some backwards compatibility with vCurrent.

### CODE REVIEW

File: packages/runtime/src/templating/custom-element.ts [hunk 1]
```diff
@@ -374,8 +374,8 @@ class HostProjector implements IViewProjector {
     return parentEncapsulationSource || this.host;
   }
 
-  public project(view: IView): void {
-    view.appendTo(this.host);
+  public project(nodes: INodeSequence): void {
+    nodes.appendTo(this.host);
```
Comment (fkleuver) [inline packages/runtime/src/templating/custom-element.ts]: This kind of thing is where the rename really shines. Love it!

File: .vscode/settings.json [hunk 2]
```diff
@@ -22,5 +22,10 @@
     "**/coverage": true,
     "**/node_modules": true,
     "**/scripts": true
-  }
+  },
+  "cSpell.words": [
+    "Renderable",
+    "attachables",
+    "bindables"
+  ]
```
Comment (fkleuver) [inline .vscode/settings.json]: So they're being squiggled as spelling errors? Are these the only words for which that happens?

**Comment by fkleuver** [comment]
Glad we had that discussion, these are some really good improvements! Let's get it in :)
You can go first, I'll fix the merge conflicts in my PR. I want to take another quick glance over my changes with these in.

File: .vscode/settings.json (hunk 2 above)
Comment (EisenbergEffect) [reply .vscode/settings.json]: I'm just adding words as I see them with squiggles. There are probably a few more.

**Pull request merged**